src/flows/pipes/RawFluxEditor/FunctionsList/DynamicFunctionsList.tsx

```diff
@@ -26,18 +26,26 @@ const sortFuncs = (a, b) => {
 }
 
 const DynamicFunctionsList: FC<Props> = ({onSelect}) => {
+  const [searchTerm, setSearchTerm] = useState('')
   const dispatch = useDispatch()
   const fluxFunctions = useSelector(getAllFluxFunctions)
+
   useEffect(() => {
     if (fluxFunctions.length === 0) {
       dispatch(getFluxPackages())
     }
   }, [])
 
-  const handleSelectItem = useCallback((func: FluxFunction) => {
-    onSelect(func)
-    event('Inject FluxDoc function into Flux Script')
-  }, [])
+  const handleSelectItem = useCallback(
+    (func: FluxFunction) => {
+      onSelect(func)
+      event('flux.function.injected', {name: `${func.package}.${func.name}`})
+      if (searchTerm) {
```

Comment:
You might wanna add this as meta data on the first event (with the name) and not make a distinct event

Reply:
I figured a separate event is needed here because user might not have searched for a function when they inject onto the editor. If `searchTem` is empty, then there wouldn't be a reason to send an empty value to events. But now I'm thinking about it, since we have a function that handles event metrics for what user searches for, it would be redundant to have it here again. In other words, if user searches for a function, then starts hovering over the filtered list to inject the function,  search term gets recorded. So, i think i'll remove it (the second event) entirely.